lib/rules/no-useless-token-range.ts

```diff
@@ -139,20 +150,22 @@ const rule = {
                   identifier.parent.property.name === 'getLastToken')),
           )
           .forEach((identifier) => {
-            const fullRangeAccess = isRangeAccess(
-              identifier.parent.parent.parent,
-            )
-              ? identifier.parent.parent.parent.parent
-              : identifier.parent.parent.parent;
+            const fullRangeAccess =
+              isMemberExpression(identifier.parent.parent.parent) &&
+              isRangeAccess(identifier.parent.parent.parent)
+                ? identifier.parent.parent.parent.parent
+                : identifier.parent.parent.parent;
             const replacementText =
               sourceCode.text.slice(
-                fullRangeAccess.range[0],
-                identifier.parent.parent.range[0],
+                fullRangeAccess.range![0],
+                identifier.parent.parent.range![0],
+              ) +
+              sourceCode.getText(
+                (identifier.parent.parent as CallExpression).arguments[0],
```

Comment:
Type check instead of casting.

Reply:
This is another example of where we're already doing the check in the filter, above, it's just not narrowing that prop downstream, so it feels redundant.  But I'll go ahead and change it. 

<img width="506" height="169" alt="image" src="https://github.com/user-attachments/assets/78a3ae23-4461-4cf0-9cb9-7e2debdd4e90" />

Reply:
Updated